velox/exec/tests/TableScanTest.cpp

```diff
@@ -1544,6 +1544,73 @@ TEST_F(TableScanTest, stringNotEqualFilter) {
       "SELECT * FROM tmp WHERE c1 != ''");
 }
 
+TEST_F(TableScanTest, arrayIsNullFilter) {
+  auto arrayType = ARRAY(BIGINT());
+  auto rowType = ROW({{"c0", arrayType}});
+  std::vector<RowVectorPtr> vectors(3);
+  auto filePaths = makeFilePaths(vectors.size());
+  for (int i = 0; i < vectors.size(); ++i) {
+    auto isNullAt = [&](vector_size_t j) {
+      return i == 0 ? false : i == 1 ? true : j % 2 != 0;
```

Comment:
This logic is hard to read. What does that do? Perhaps, add a comment or re-write for clarity.